app/models/manageiq/providers/redhat/infra_manager/ovirt_services/strategies/v4.rb

```diff
@@ -607,15 +607,22 @@ def update_vm_memory(vm, vm_service, memory)
       virtual = calculate_adjusted_virtual_memory(vm, memory)
       guaranteed = calculate_adjusted_guaranteed_memory(vm, memory)
 
+      # The required memory cannot exceed the max configured memory of the VM. Therefore, we'll increase the max
+      # memory up to 1TB or to the required limit, to allow a successful update for the VM.
+      # Once 'max' memory attribute will be introduced, this code should be replaced with the specified max memory.
+      supports_max = ext_management_system.version_at_least?('4.1')
+      max = calculate_max_memory(vm, memory) if supports_max
+
       # If the virtual machine is running we need to update first the configuration that will be used during the
       # next run, as the guaranteed memory can't be changed for the running virtual machine.
       if vm.status == OvirtSDK4::VmStatus::UP
         vm_service.update(
           OvirtSDK4::Vm.new(
             :memory        => virtual,
             :memory_policy => {
-              :guaranteed => guaranteed
-            }
+              :guaranteed => guaranteed,
+              :max        => (max if supports_max)
```

Comment:
With this code if `supports_max` is `false` the value here will be `nil`. Should we not set it at all here?

Reply:
the `.compact` at the end of the hash will remove it from the map if the value is null.
The alternative is extracting it into a local variable and add 'max' conditionally which will add complexity to this method.